op-chain-ops/cmd/check-canyon/main.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-service/client"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/sources"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethdb/memorydb"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/trie"
+)
+
+func PreCanyonEncode(receipts types.Receipts) [][]byte {
```

Comment:
There isnt a lower level function from `op-geth` that does this operation?

Reply:
The `receipts.EncodeIndex` is that, but here it does this outer loop to force all receipts to be a certain way, to emulate what it would be pre or post canyon.

Reply:
Not really. It's kinda ugly, but no one I know of does MPT proofs against the receipts/transaction root